<?xml version="1.0"?>
<!--|
    Copyright (C) 2019-2022 Learning Algorithms and Systems Laboratory, EPFL, Switzerland
    Authors:  Konstantinos Chatzilygeroudis (maintainer)
              Matthias Mayr
              Bernardo Fichera
    email:    costashatz@gmail.com
              matthias.mayr@cs.lth.se
              bernardo.fichera@epfl.ch
    Other contributors:
              Yoan Mollard (yoan@aubrune.eu)
              Walid Amanhoud (walid.amanhoud@epfl.ch)
    website:  lasa.epfl.ch

    This file is part of iiwa_ros.

    iiwa_ros is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    iiwa_ros is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

|-->
<launch>
    <!-- Select the robot -->
    <arg name="robot_name" default="iiwa"/>
    <arg name="model" default="14" />

    <!-- Select the controller -->
    <arg name="controller" default="TorqueController"/>

    <!-- Select the booleans grabber and force sensor if we want them mounted on the robot -->
    <arg name="force_sensor" default="False"/>
    <arg name="grabber" default="False"/>

    <!-- Select the hardware interface -->
    <arg if="$(eval controller == 'TorqueController')" name="hardware_interface" value="EffortJointInterface"/>
    <arg if="$(eval controller == 'PositionTorqueController')" name="hardware_interface" value="EffortJointInterface"/>
    <arg if="$(eval controller == 'PositionController')" name="hardware_interface" value="PositionJointInterface"/>
    <arg if="$(eval controller == 'PositionTrajectoryController')" name="hardware_interface" value="PositionJointInterface"/>
    <!-- By default assume EffortJointInterface -->
    <arg unless="$(eval (controller == 'PositionController') or (controller == 'PositionTorqueController') or (controller == 'TorqueController') or (controller == 'PositionTrajectoryController'))" name="hardware_interface" value="EffortJointInterface"/>

    <!-- Upload iiwa URDF -->
    <include file="$(find iiwa_description)/launch/iiwa$(arg model)_upload.launch">
        <arg name="hardware_interface" value="$(arg hardware_interface)"/>
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="force_sensor" value="$(arg force_sensor)"/>
        <arg name="grabber" value="$(arg grabber)"/>
    </include>

    <!-- Spawn iiwa service -->
    <include file="$(find iiwa_tools)/launch/iiwa_service.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
    </include>
</launch>
